base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] Please make sure to add the association between the Namebale and BaseDecorator classes to ensure correct readability and accessibility between the two classes. 🙏 
`attr_accessor :nameable`